### DISCUSSION TITLE
Issue when migrating from azure devops nested templates

### DISCUSSION THREAD

**Discussion by javierpe27** [github/gh-actions-importer#293] [Feedback]
Hello,

I have 3 pipelines, one is the main pipeline that uses one as a template and this second one calls a third as a template ; Main -> Template 1 -> Template 2
The templates are in the same repo (in a different folder), however, I'm having issues using the automation to migrate these pipelines receiving an error that the template with the provided path doesn't exist, is there any special way I should provide the path? I've tried with relative and absolute paths but it still doesn't work. Any idea?

**Comment by aveva-matthewngor** [comment]
Hi, I believe I'm running into a similar issue, except the workflow references a template located in a specific subfolder of another repository within the ADO org. The migration/import process fails to resolve the template, even though the path and repository are correct and accessible. Branch references are verified. I don't believe it's a permission issue as the PATs used have full access to the ADO org. I see in the [GitHub Actions docs](https://docs.github.com/en/actions/how-tos/migrating-to-github-actions/using-github-actions-importer-to-automate-migrations/migrating-from-azure-devops-with-github-actions-importer#supported-templates) that templates in a different ADO repo is supported by the GitHub Actions Importer. During the audit command, I see errors like:

- Message: The template with the following path does not exist: (pipelines/example-template.yml@some-other-repo).
- Message: Resource not found. (GET 404): Not Found: (https://dev.azure.com/ORG/PROJECT/_apis/git/repositories/REPO/items?api-version=5.0-preview&path=pipelines/templates/pipelines/templates/another-example-template.yml). 

Note how the path segment `pipelines/templates` in the second error is repeated. I think this duplication is causing the template resolution to fail. 

I've tried adding a [credentials file](https://docs.github.com/en/actions/reference/supplemental-arguments-and-settings?search-overlay-input=My+credentials+passed+in+to+credentials-file+flag+is+correct.+Why+am+I+still+seeing+a+404+resource+not+found+error+when+doing+gh+actions-importer+audit&search-overlay-ask-ai=true#using-a-credentials-file-for-authentication) that just contains a few lines of code (The ADO org url, access_token, and provider parameters), but I still run into error messages, even when specifying the "azure_devops" `provider` parameter. When including this file, but excluding the `provider` parameter, I see a different error message: "There was an error retrieving file contents from GitHub(). Unable to parse GitHub repository URL. Ensure it is in the format of https://github.com/:owner_name/:repo_name." I tried adding the respective parameters for the destination GitHub org in the credentials file, and migrating over the dependent repository into GitHub to see if the migration tool is actually attempting to resolve the dependency in GitHub, but this didn't resolve the errors either.

Do you have any suggestions or other things I could try? Is this actually a limitation of the GitHub Actions Importer, or could I be missing some additional configuration step for this scenario?

Any guidance or best practices would be appreciated. Thanks!

**Comment by akordowski** [comment]
Same issue here.
